The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5925719656/job/16065680696) [(plain)](https://github.com/rust-lang/rust/commit/82f4f835e01d4ceb0456d241a47066c79b90f36c/checks/16065680696/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=waywardmonkeys
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_943ec7a5-c232-4ee4-98a8-60d2814a0bea
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=augment-type-keyword-docs
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_943ec7a5-c232-4ee4-98a8-60d2814a0bea
GITHUB_REF=refs/pull/115053/merge
GITHUB_REF_NAME=115053/merge
GITHUB_REF_PROTECTED=false
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=b425e4209ee94e25dd4a89050eaaf8329d199ad8
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_943ec7a5-c232-4ee4-98a8-60d2814a0bea
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_943ec7a5-c232-4ee4-98a8-60d2814a0bea
GITHUB_TRIGGERING_ACTOR=waywardmonkeys
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/115053/merge
GITHUB_WORKFLOW_SHA=b425e4209ee94e25dd4a89050eaaf8329d199ad8
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_18_X64=/opt/hostedtoolcache/go/1.18.10/x64
---
---- library/std/src/keyword_docs.rs - type_keyword (line 1843) stdout ----
error: where clauses are not allowed after the type for type aliases
##[error] --> library/std/src/keyword_docs.rs:1845:28
  |
5 | type TypeAlias<T> = Bar<T> where T: Foo;
  |
  = note: see issue #89122 <https://github.com/rust-lang/rust/issues/89122> for more information

error[E0412]: cannot find type `Arc` in this scope
error[E0412]: cannot find type `Arc` in this scope
##[error] --> library/std/src/keyword_docs.rs:1844:20
  |
4 | type ArcMutex<T> = Arc<Mutex<T>>;
  |
help: consider importing one of these items
  |
3 + use alloc::sync::Arc;
3 + use alloc::sync::Arc;
  |
3 + use std::sync::Arc;
  |

error[E0412]: cannot find type `Mutex` in this scope
##[error] --> library/std/src/keyword_docs.rs:1844:24
  |
4 | type ArcMutex<T> = Arc<Mutex<T>>;
  |                        ^^^^^ not found in this scope
error: doctest failed, to rerun pass `-p std --doc`
  |
help: consider importing this struct
  |
3 + use std::sync::Mutex;
3 + use std::sync::Mutex;
  |

error[E0405]: cannot find trait `Foo` in this scope
##[error] --> library/std/src/keyword_docs.rs:1845:37
  |
5 | type TypeAlias<T> = Bar<T> where T: Foo;

error[E0412]: cannot find type `Bar` in this scope
##[error] --> library/std/src/keyword_docs.rs:1845:21
  |
  |
5 | type TypeAlias<T> = Bar<T> where T: Foo;

error: aborting due to 5 previous errors

Some errors have detailed explanations: E0405, E0412.
```

</details>

--- a/library/std/src/keyword_docs.rs
+++ b/library/std/src/keyword_docs.rs
@@ -1820,7 +1820,7 @@ mod true_keyword {}
 
 #[doc(keyword = "type")]
 //
-/// Define an alias for an existing type.
+/// Define an [alias] for an existing type.
 ///
 /// The syntax is `type Name = ExistingType;`.
 ///
@@ -1836,6 +1836,13 @@ mod true_keyword {}
 /// let k: Kilograms = 3;
 ///
 /// assert_eq!(m, k);
+/// ```
+///
+/// A type can be generic:
+///
+/// ```rust
+/// # use std::sync::{Arc, Mutex};
+/// type ArcMutex<T> = Arc<Mutex<T>>;
 /// ```
 ///
 /// In traits, `type` is used to declare an [associated type]:
@@ -1860,6 +1867,7 @@ mod true_keyword {}
 ///
 /// [`trait`]: keyword.trait.html
 /// [associated type]: ../reference/items/associated-items.html#associated-types
+/// [alias]: ../reference/items/type-aliases.html
 mod type_keyword {}
 
 #[doc(keyword = "unsafe")]

